tools/oeedger8r/CMakeLists.txt

```diff
@@ -1,6 +1,76 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+find_program(OCAMLOPT ocamlopt)
+
+if (OCAMLOPT)
+  set(BINARY ${CMAKE_CURRENT_BINARY_DIR}/oeedger8r${CMAKE_EXECUTABLE_SUFFIX})
+  add_custom_command(
+    OUTPUT ${BINARY}
+    # Copy sources to build folder.
+    COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/intel/*
+            ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/src/*
+            ${CMAKE_CURRENT_BINARY_DIR}
+    # Generate lexer and parser.
+    COMMAND ocamllex Lexer.mll
+    COMMAND ocamlyacc Parser.mly
+    # Generate Intel library.
+    # Use -w -3 to suppress
+    # warning 3: deprecated: String.lowercase
+    COMMAND
+      ocamlopt -c -bin-annot -for-pack Intel -w -3 Ast.ml Util.ml SimpleStack.ml
+      Plugin.ml Preprocessor.ml Parser.mli Parser.ml Lexer.ml CodeGen.ml
+    COMMAND ocamlopt -pack -o Intel.cmx Ast.cmx Util.cmx SimpleStack.cmx
+            Plugin.cmx Preprocessor.cmx Parser.cmx Lexer.cmx CodeGen.cmx
+    # Generate oeedger8r
+    # Additionally suppress
+    # Warning 40: input_files was selected from type Intel.Util.edger8r_params.
+    # It is not visible in the current scope, and will not
+    # be selected if the type becomes unknown.
+    COMMAND
+      ocamlopt -bin-annot str.cmxa unix.cmxa Intel.cmx -w -3-40 Common.mli
+      Common.ml Headers.mli Headers.ml Sources.mli Sources.ml Emitter.ml main.ml
+      -o ${BINARY}
+    DEPENDS src/dune
+            src/Common.mli
```

Comment:
Shouldn't this be `${CMAKE_CURRENT_SOURCE_DIR}/src/Common.mli` (same for the following lines)?

Reply:
cmake by default will prefix relative paths with the current source directory if a file exists; if a valid file does not exist, then it will prefix it with current binary directory. I feel omitting the ${CMAKE_CURRENT_SOURCE_DIR} prefix is less verbose.